src/front/spv/function.rs

```diff
@@ -157,19 +165,128 @@ impl<I: Iterator<Item = u32>> super::Parser<I> {
 
         fun.body = flow_graph.to_naga()?;
 
+        // done
+        let fun_handle = module.functions.append(fun);
         match self.lookup_entry_point.remove(&fun_id) {
             Some(ep) => {
+                // create a wrapping function
```

Comment:
So these steps done for SPIRV are basically, when having an entrypoint, create a function that gets called immediately as first instruction in the entrypoint execution. This function initializes the arguments that are "passed" into the entrypoint of SPIRV, so they are usable within the entrypoint function?

Reply:
The problem is that SPIRV's functions would try to access the inputs/outputs as globals (in the relevant Input/Output storage classes). We aren't trying to solve this at the function level. Instead, we solve it at the entry point (and module) level: we create the globals for inputs/outputs, but they are just `Private` globals (so, different per invocation). The wrapping function for an entry point is needed to populate these globals before we call the 1:1 translated function from SPIR-V.